### ISSUE TITLE
clojure.core/load

### ISSUE THREAD

**Issue by semperos** [borkdude/clj-kondo#798]
**version**

```
v2020.02.28
```

**platform**

* Native executable (from GitHub release)
* macOS 10.15.2

**editor**

N/A

**problem**

Clojure's `load` can introduce new vars into the current namespace, but clj-kondo reports such symbols as undefined in the calling namespace.

**repro**

If we have two files, one of which `load`s the other:

```clj
;; repro_secret_sauce.clj

(def secret-sauce "Mustard")
```

```clj
;; repro.clj

(ns repro)

(load "repro_secret_sauce")

(println secret-sauce) ;; <-- clj-kondo reports as undefined
```

When linted:

```
$ clj-kondo --lint repro.clj
repro.clj:5:10: error: unresolved symbol secret-sauce
```

But when evaluated by Clojure:

```
$ clj -Scp "$(clj -Spath):." repro.clj
Mustard
```

**expected behavior**

If possible, the most optimal behavior in my opinion would be:

Upon encountering a `load` call, clj-kondo analyzes the resource passed to `load` and establishes its public vars as public vars of the calling namespace (emulating what Clojure does). In the repro case described above, `secret-sauce` would be defined and no warnings/errors emitted by clj-kondo.

If not feasible, then either a clj-kondo error, or a warning coupled with an option to either discontinue a file's analysis when `load` is encountered, or be able to have clj-kondo disengage the "unresolved symbol" linter for the remainder of such a file.

**Thank you for your open source contributions!** I am happy to attempt a PR if there is an acceptable approach to solving it from your perspective.

**Comment by borkdude** [comment]
Clj-kondo does not go out on its own to lint other files from disk unless you explicitly feed those files in via the `--lint` args. You can use `declare` to declare vars that are brought in by the call to load.
